LearnBoost/shorty
# Update Node to 6.13.0 LTS + current lib versions

This PR does the following:

- Updates recommendation for Node to the 6.x LTS: 6.13.0
- Updates all dependencies to their newest versions
- Swaps Jade for Pug, which is the newer library
- Basic code updates to update syntax for Express v4
- Adds Express libraries that were split off from core that now must be included explicitly
- Removes the /stats page
- Removes basic auth (as that was basically protecting the stats page)
- Removes Socket.io functionality, as we don't need real-time stats anymore

How to test:

- Install and use Node 6.13.0 (nvm makes this easy)
- Run `npm install`
- Run `npm start`
- Visit `http://localhost:3000` -> you should see a simple HTML page
- Run `curl -X POST -d "url=https://www.learnboost.com" localhost:3000` from another terminal
- This will return a url like: `https://lrn.cc/0` You should be able to visit `http://localhost:3000/0` to verify your redirect worked.
